src/sentry/services/hybrid_cloud/organization_mapping/impl.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import fields
+from typing import Any, Optional
+
+from django.db import IntegrityError, transaction
+
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.user import User
+from sentry.services.hybrid_cloud.organization_mapping import (
+    APIOrganizationMapping,
+    OrganizationMappingService,
+)
+
+
+class DatabaseBackedOrganizationMappingService(OrganizationMappingService):
+    def create(
+        self,
+        *,
+        user: User,
+        organization_id: int,
+        slug: str,
+        region_name: str,
+        idempotency_key: Optional[str] = "",
+        # There's only a customer_id when updating an org slug
+        customer_id: Optional[str] = None,
+    ) -> APIOrganizationMapping:
+        try:
+            with transaction.atomic():
```

Comment:
Is this transaction necessary?  There seems to be only one operation, which with Django would implicitly be its own singular transaction.

Reply:
If an integrity error is raised, without the transaction snapshot here further queries would fail.

Reply:
Oh right.  Man Django ORM is awful!!!!